source/common/network/listen_socket_impl.h

```diff
@@ -33,6 +40,23 @@ class ListenSocketImpl : public SocketImpl {
   void setupSocket(const Network::Socket::OptionsSharedPtr& options);
   void setListenSocketOptions(const Network::Socket::OptionsSharedPtr& options);
   Api::SysCallIntResult bind(Network::Address::InstanceConstSharedPtr address) override;
+
+  void close() override {
+    RELEASE_ASSERT(io_handle_ != nullptr,
```

Comment:
Sorry for being dense, but I don't understand why these checks are here. Either just remove them or add null checking here and below?

/wait

Reply:
Sorry, can you offer an example of `null checking`?

Reply:
done. replaced by if (nullptr != xxx) so that envoy won't crash